api/citations/utils.py

```diff
@@ -0,0 +1,29 @@
+import os
+
+from citeproc import CitationStylesStyle, CitationStylesBibliography
+from citeproc import Citation, CitationItem
+from citeproc import formatter
+from citeproc.source.json import CiteProcJSON
+
+from website.settings import CITATION_STYLES_PATH
+
+
+def render_citation(node, style='apa'):
+    """Given a node, return a citation"""
+    data = [node.csl, ]
+
+    bib_source = CiteProcJSON(data)
+
+    bib_style = CitationStylesStyle(os.path.join(CITATION_STYLES_PATH, style), validate=False)
+
+    bibliography = CitationStylesBibliography(bib_style, bib_source, formatter.plain)
+
+    citation = Citation([CitationItem(node._id)])
+
+    bibliography.register(citation)
+
+    def warn(citation_item):
+        pass
+
+    bibliography.cite(citation, warn)
```

Comment:
Trivial, but if you're passing a noop as a error-handling callback, `lambda x: None` should work just as well, and obviates the need to define one above.
```
In [213]: w = lambda x: None

In [214]: def p(x):
   ....:     pass
   ....:

In [215]: p('') == w('')
Out[215]: True
```